### DISCUSSION TITLE
Ref name passed as function parameter not found

### DISCUSSION THREAD

**Discussion by Regenhardt** [typst/typst#4818] [General]
### Description

I have a function that creates a figure from multiple strings by constructing image tags from them. This function also accepts a string argument that it uses to create a reference to this created figure. 

Unfortunately, the compiler won't properly find this reference, even though it seems some other part of typst (or maybe tinymist?) does find the figure.

This is my situation basically:
```typst
#let cmimg(modelName) = image("img/results/confmat90/" + modelName + "_50e_cm.png")

#let cms(caption: [Add caption content here], refName: [my_ref], ..modelNames) = block(breakable: false)[#figure(
  grid(
    columns: 2, 
    gutter: 2mm, 
    ..modelNames.pos().map(model => cmimg(model))
  ), 
  caption: caption
) <#refName>]

#cms(
  "1c_8f_1l_10u", 
  "1c_8f_2l_10u_5u", 
  "2c_16f_8f_1l_10u", 
  "2c_16f_8f_2l_10u_5u",
  caption: [Konfusionsmatrizen der kleinen Modelle nach Training über 50 Epochen: Open links je eine Conv- und LSTM-Schicht , oben rechts eine Conv-Schicht und zwei LSTM-Schichten, unten links zwei Conv-Schichten und eine LSTM-Schicht, unten rechts je zwei Schichten.],
  refName: [small_models_confmat]
)

Wie in @small_models_confmat zu sehen ist, ...
```

As you can see in the following screenshot, it doesn't find the label but does find the label at the same time, as it displays an error message but also displays the figure's caption in the popup:
![grafik](https://github.com/user-attachments/assets/91eb3f4e-8eeb-4dd7-a895-2cbe2285b8c0)

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Interpolating a variable in a label `<#refName>` is not a thing. It creates a label with the name literally `#refName`. That's why it doesn't work.

What you want to do is, replace `<#refName>` with `#label(refName)`. You'll also need to change `refName: [small_models_confmat]` to `refName: "small_models_confmat"` because the `label` function takes a string rather than content.

**Reply by Regenhardt** [reply]
Wow that was fast, thanks a lot! I guess it was indeed just tinymist being too optimistic before.
